src/scripting/lua_formula_bridge.cpp

```diff
@@ -70,7 +70,10 @@ class lua_callable : public formula_callable {
 		add_input(inputs, "__list");
 		add_input(inputs, "__map");
 		for(lua_pushnil(mState); lua_next(mState, table_i); lua_pop(mState,1)) {
-			if(lua_isstring(mState, -2) && !lua_isnumber(mState, -2)) {
+			lua_pushvalue(mState, -2);
+			bool t = (lua_type(mState, -1) == LUA_TSTRING) && !lua_isnumber(mState, -1);
+			lua_pop(mState, 1);
+			if (t) {
```

Comment:
I suspect the push+pop actually isn't necessary, unless I misread something in the Lua manual.

Also, using this method it's clearly redundant to test for both string and number. Perhaps it was also redundant in the old method, if `lua_isnumber` tests for convertability in the same way that `lua_isstring` does; I probably did this in some misguided attempt to work around the behaviour of `lua_isstring`. In any case, testing for string only should be sufficient.

Reply:
I cannot change the index. So I push a copy and I need to clean that up, so a pop. Taht way, the lua_isnumber doesn't change the string "1" to the number 1 and cause a crash.

I decided to do this instead of lua_type because TSTRING implies !TNUMBER and so we must be doing the is-a-string-but-not-convertible-to-a-number thing for some reason.

Reply:
Hmm. I'm not quite sure whether this is what I originally intended, or if I was trying to work around `lua_isstring` also accepting numbers. Thinking about it now though, it could (in rare cases) make sense to include strings that are convertible to numbers, as these could be valid WML keys.